### PR TITLE
Fix various warnings with latest gcc

### PR DESCRIPTION
**Pull request by teto**
When updating ubuntu, gcc5 got installed and would stop compiling DCE because of some warnings.

### CODE REVIEW

**Comment by thehajime** [comment]
that would be nice to have this patch.
which revision of ubuntu are you using ?

for the ubuntu specific case, while I don't plan to support/test with x.10 release (i.e., 15.10), this exercise  will be help for a future LTS release (x.04).

**Comment by teto** [comment]
I am using 15.10. gcc seems to be a lot smarter, reasoning on ranges etc... thus the fixes make sense even for older compilers.

**Comment by thehajime** [comment]
Please add the version number of gcc and Ubuntu in the commit message for the reference.

File: model/local-datagram-socket-fd.cc [hunk 1]
```diff
@@ -888,6 +888,9 @@ LocalDatagramSocketFd::RemoveConnected (LocalDatagramSocketFd *freeOne, bool and
           }
       }
       break;
+    default:
+        NS_LOG_WARN("Unhandled case");
```
Comment (thehajime) [inline model/local-datagram-socket-fd.cc]: add a space before a bracket '('.

File: test/netlink-socket-test.cc [hunk 2]
```diff
@@ -434,7 +434,7 @@ void
 NetlinkSocketTestCase::ReceiveMulticastPacket (Ptr<Socket> socket)
 {
   Ptr<Packet> packet;
-  while (packet = socket->Recv ())
+  while ( (packet = socket->Recv ()))
```
Comment (thehajime) [inline test/netlink-socket-test.cc]: remove a space between '(' and '('.

**Comment by teto** [comment]
Fixed

**Comment by thehajime** [comment]
thanks, merged.

**Pull request merged**